exasol_advanced_analytics_framework/query_handler/context/scope_query_handler_context.py

```diff
@@ -1,9 +1,33 @@
-from abc import abstractmethod
+import enum
+from abc import abstractmethod, ABC
 
 from exasol_advanced_analytics_framework.query_handler.context.query_handler_context import QueryHandlerContext
 from exasol_advanced_analytics_framework.query_handler.context.proxy.object_proxy import ObjectProxy
 
 
+class Connection(ABC):
+
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        pass
```

Comment:
If there is need to communicate additional information about the interface, I personally prefer the doc string based empty definition.

```suggestion
    @property
    @abstractmethod
    def name(self) -> str:
        """ <some comment/docstring> ""
```